pennylane_lightning_kokkos/src/algorithms/ObservablesKokkos.hpp

```diff
@@ -0,0 +1,521 @@
+
+    /**
+     * @brief Test whether this object is equal to another object
+     */
+    [[nodiscard]] bool operator==(const ObservableKokkos<T> &other) const {
+        return typeid(*this) == typeid(other) && isEqual(other);
+    }
+
+    /**
+     * @brief Test whether this object is different from another object.
+     */
+    [[nodiscard]] bool operator!=(const ObservableKokkos<T> &other) const {
+        return !(*this == other);
+    }
+};
+
+/**
+ * @brief Class models named observables (PauliX, PauliY, PauliZ, etc.)
+ *
+ * @tparam T Floating point type
+ */
+template <typename T> class NamedObsKokkos final : public ObservableKokkos<T> {
+  private:
+    std::string obs_name_;
+    std::vector<size_t> wires_;
+    std::vector<T> params_;
+
+    [[nodiscard]] bool
+    isEqual(const ObservableKokkos<T> &other) const override {
+        const auto &other_cast = static_cast<const NamedObsKokkos<T> &>(other);
+
+        return (obs_name_ == other_cast.obs_name_) &&
+               (wires_ == other_cast.wires_) && (params_ == other_cast.params_);
+    }
+
+  public:
+    /**
+     * @brief Construct a NamedObsKokkos object, representing a given
+     * observable.
+     *
+     * @param obs_name Name of the observable.
+     * @param wires Argument to construct wires.
+     * @param params Argument to construct parameters
+     */
+    NamedObsKokkos(std::string obs_name, std::vector<size_t> wires,
+                   std::vector<T> params = {})
+        : obs_name_{std::move(obs_name)}, wires_{std::move(wires)},
+          params_{std::move(params)} {}
+
+    [[nodiscard]] auto getObsName() const -> std::string override {
+        using Pennylane::Util::operator<<;
+        std::ostringstream obs_stream;
+        obs_stream << obs_name_ << wires_;
+        return obs_stream.str();
+    }
+
+    [[nodiscard]] auto getWires() const -> std::vector<size_t> override {
+        return wires_;
+    }
+
+    void applyInPlace(StateVectorKokkos<T> &sv) const override {
+        sv.applyOperation(obs_name_, wires_, false, params_);
+    }
+};
+
+/**
+ * @brief Class models arbitrary Hermitian observables.
+ */
+template <typename T>
+class HermitianObsKokkos final : public ObservableKokkos<T> {
+  public:
+    using MatrixT = std::vector<std::complex<T>>;
+
+  private:
+    std::vector<std::complex<T>> matrix_;
+    std::vector<size_t> wires_;
+    inline static const Util::MatrixHasher mh;
+
+    [[nodiscard]] bool
+    isEqual(const ObservableKokkos<T> &other) const override {
+        const auto &other_cast =
+            static_cast<const HermitianObsKokkos<T> &>(other);
+
+        return (matrix_ == other_cast.matrix_) && (wires_ == other_cast.wires_);
+    }
+
+  public:
+    /**
+     * @brief Create Hermitian observable
+     *
+     * @param matrix Matrix in row major format.
+     * @param wires Wires the observable applies to.
+     */
+    // template <typename T1>
```

Comment:
Is this comment necessary?

Reply:
Thanks @AmintorDusko ! Good catch. Let me remove it.